pkg/controllers/cluster/import.go

```diff
@@ -139,6 +140,21 @@ func (i *importHandler) deleteOldAgent(cluster *fleet.Cluster, kc kubernetes.Int
 		}
 	}
 
+	if i.systemNamespace != config.DefaultNamespace {
+		// Attempt to delete the agent in the default namespace if it has changed.
+		if err := kc.AppsV1().Deployments(config.DefaultNamespace).Delete(i.ctx, config.AgentConfigName, metav1.DeleteOptions{}); err != nil {
```

Comment:
We need to check if deployment or namespace exist and if they don't exist we need to skip. Otherwise this will fail.

Reply:
Good catch. Worked for me because it existed